### ISSUE TITLE
Password Generator advanced settings are applied even after advanced is disabled

### ISSUE THREAD

**Issue by JVimes** [keepassxreboot/keepassxc#10169]
## Overview

"Pick characteres from every group" is selected by default but seemingly not honored before Advanced is shown the first time.

https://github.com/keepassxreboot/keepassxc/assets/6148201/6f58004e-cbf2-441c-aa0f-a3bcea3c3604

## Steps to Reproduce
1. Rename or delete keepassxc.ini, if it exists
2. Open password generator, set length to 10, turn off special characters
3. Regenerate several times

## Expected Behavior

Each selected character group is included.

## Actual Behavior

Eventually I find a password without numbers.

## Context

- I ran into this making a password for a site that requires numbers.
- Things seem to work as expected after Advanced is shown the first time, even after closing it, but it's hard to be sure.
- I'm glad "pick characters from every group" is enabled by default.

KeePassXC - Version 2.7.6
Revision: dd21def

Qt 5.15.10
Debugging mode is disabled.

Operating system: Windows 10 Version 2009
CPU architecture: x86_64
Kernel: winnt 10.0.19045

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock

Cryptographic libraries:
- Botan 2.19.3

Operating System: Windows 10 22H2 (build 19045.3803)

**Comment by droidmonkey** [comment]
To use advanced features of password generator the advanced button must be selected.

**Comment by JVimes** [comment]
Wow

**Comment by JVimes** [comment]
@droidmonkey You misunderstood. Should I try explain or log this again with different wording?

**Comment by droidmonkey** [comment]
To use the settings under the advanced portion of the password generator, the advanced button must be selected and the advanced settings visible. In basic mode, none of those settings apply.

**Comment by JVimes** [comment]
@droidmonkey Sorry, I could have been clearer. I believe:

- "Pick characters from every group" should be the default non-Advanced behavior, but isn't on a clean install (no .ini file).
- I'm seeing Advanced settings affect non-Advanced mode.  "Pick characters from every group" affects non-Advanced mode, but only after Advanced has been opened-and-closed at least once on that machine.

**Comment by droidmonkey** [comment]
You are absolutely correct. There is a bug where advanced settings are applied even after deselected. Sorry about that!

**Comment by JVimes** [comment]
Thank you! Have a great evening.
